File: binding/src/printsupport/libqabstractprintdialog.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qabstractprintdialog.html
pub const qabstractprintdialog = struct {

    /// New constructs a new QAbstractPrintDialog object.
    pub fn New(printer: ?*anyopaque) QtC.QAbstractPrintDialog {
        return qtc.QAbstractPrintDialog_new(@ptrCast(printer));
    }


    /// New2 constructs a new QAbstractPrintDialog object.
    pub fn New2(printer: ?*anyopaque, parent: ?*anyopaque) QtC.QAbstractPrintDialog {
        return qtc.QAbstractPrintDialog_new2(@ptrCast(printer), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAbstractPrintDialog_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qabstractprintdialog.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetOptionTabs(self: ?*anyopaque, tabs: []const u8) void {
        qtc.QAbstractPrintDialog_SetOptionTabs(@ptrCast(self), @ptrCast(tabs));
    }

    pub fn SetPrintRange(self: ?*anyopaque, range: i32) void {
        qtc.QAbstractPrintDialog_SetPrintRange(@ptrCast(self), @intCast(range));
    }

    pub fn PrintRange(self: ?*anyopaque, ) i32 {
        return qtc.QAbstractPrintDialog_PrintRange(@ptrCast(self));
    }

    pub fn SetMinMax(self: ?*anyopaque, min: i32, max: i32) void {
        qtc.QAbstractPrintDialog_SetMinMax(@ptrCast(self), min, max);
    }

    pub fn MinPage(self: ?*anyopaque, ) i32 {
        return qtc.QAbstractPrintDialog_MinPage(@ptrCast(self));
    }

    pub fn MaxPage(self: ?*anyopaque, ) i32 {
        return qtc.QAbstractPrintDialog_MaxPage(@ptrCast(self));
    }

    pub fn SetFromTo(self: ?*anyopaque, fromPage: i32, toPage: i32) void {
        qtc.QAbstractPrintDialog_SetFromTo(@ptrCast(self), fromPage, toPage);
    }

    pub fn FromPage(self: ?*anyopaque, ) i32 {
        return qtc.QAbstractPrintDialog_FromPage(@ptrCast(self));
    }

    pub fn ToPage(self: ?*anyopaque, ) i32 {
        return qtc.QAbstractPrintDialog_ToPage(@ptrCast(self));
    }

    pub fn Printer(self: ?*anyopaque, ) QtC.QPrinter {
        return qtc.QAbstractPrintDialog_Printer(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAbstractPrintDialog_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qabstractprintdialog.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAbstractPrintDialog_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qabstractprintdialog.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAbstractPrintDialog_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/abstractprintdialog.html#types
pub const enums = struct {
    pub const PrintRange = enum {
        pub const AllPages: i32 = 0;
        pub const Selection: i32 = 1;
        pub const PageRange: i32 = 2;
        pub const CurrentPage: i32 = 3;
    };

    pub const PrintDialogOption = enum {
        pub const PrintToFile: i32 = 1;
        pub const PrintSelection: i32 = 2;
        pub const PrintPageRange: i32 = 4;
        pub const PrintShowPageSize: i32 = 8;
        pub const PrintCollateCopies: i32 = 16;
        pub const PrintCurrentPage: i32 = 64;
    };

};

File: binding/src/printsupport/libqabstractprintdialog.cpp
#include <QAbstractPrintDialog>
#include <QDialog>
#include <QPrinter>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QWidget>
#include <qabstractprintdialog.h>
#include "libqabstractprintdialog.h"
#include "libqabstractprintdialog.hxx"

QAbstractPrintDialog* QAbstractPrintDialog_new(QPrinter* printer) {
	 return new QAbstractPrintDialog(printer);
}

QAbstractPrintDialog* QAbstractPrintDialog_new2(QPrinter* printer, QWidget* parent) {
	 return new QAbstractPrintDialog(printer, parent);
}

libqt_string QAbstractPrintDialog_Tr(const char* s) {
	QString _ret = QAbstractPrintDialog::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QAbstractPrintDialog_SetOptionTabs(QAbstractPrintDialog* self, const libqt_list tabs) {
	self->setOptionTabs(*tabs);
}

void QAbstractPrintDialog_SetPrintRange(QAbstractPrintDialog* self, int range) {
	self->setPrintRange(static_cast<QPrinter::PrintRange>(range));
}

int QAbstractPrintDialog_PrintRange(const QAbstractPrintDialog* self) {
	return self->printRange();
}

void QAbstractPrintDialog_SetMinMax(QAbstractPrintDialog* self, int min, int max) {
	self->setMinMax(min, max);
}

int QAbstractPrintDialog_MinPage(const QAbstractPrintDialog* self) {
	return self->minPage();
}

int QAbstractPrintDialog_MaxPage(const QAbstractPrintDialog* self) {
	return self->maxPage();
}

void QAbstractPrintDialog_SetFromTo(QAbstractPrintDialog* self, int fromPage, int toPage) {
	self->setFromTo(fromPage, toPage);
}

int QAbstractPrintDialog_FromPage(const QAbstractPrintDialog* self) {
	return self->fromPage();
}

int QAbstractPrintDialog_ToPage(const QAbstractPrintDialog* self) {
	return self->toPage();
}

QPrinter* QAbstractPrintDialog_Printer(const QAbstractPrintDialog* self) {
	return self->printer();
}

libqt_string QAbstractPrintDialog_Tr2(const char* s, const char* c) {
	QString _ret = QAbstractPrintDialog::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QAbstractPrintDialog_Tr3(const char* s, const char* c, int n) {
	QString _ret = QAbstractPrintDialog::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QAbstractPrintDialog_Delete(QAbstractPrintDialog* self) {
    delete self;
}

